README.md

```diff
@@ -46,70 +38,172 @@ Maven:
 
 ## Usage
 
-Instantiate the task.
+To create the task instance, you need to use the `WebRTCTaskBuilder` instance
+which can be used to configure the task before creating it.
+
+The below configuration represents the default values chosen by the builder as
+if you had not configured the builder and just called `.build()` directly.
 
 ```java
-final WebRTCTask task = new WebRTCTask();
+final WebRTCTask task = new WebRTCTaskBuilder()
+    .withVersion(WebRTCTaskVersion.V1)
+    .withHandover(true)
+    .withMaxChunkLength(262144)
+    .build();
 ```
 
-Then, register a message handler:
+To send offers, answers and candidates, use the following task methods:
+
+* `task.sendOffer(offer: @NonNull Offer): void`
+* `task.sendAnswer(answer: @NonNull Answer): void`
+* `task.sendCandidates(candidate: @NonNull Candidate[]): void`
+
+You can register an event handler in the following way:
 
 ```java
 task.setMessageHandler(new MessageHandler() {
     @Override
-    public void onOffer(SessionDescription sd) {
+    public void onOffer(@NonNull Offer offer) {
         // Handle offer
     }
 
     @Override
-    public void onAnswer(SessionDescription sd) {
+    public void onAnswer(@NonNull Answer answer) {
         // Handle answer
     }
 
     @Override
-    public void onCandidates(List<IceCandidate> candidates) {
-        for (IceCandidate candidate : candidates) {
-            peerConnection.addIceCandidate(candidate);
-        }
+    public void onCandidates(@NonNull Candidate[] candidates) {
+        // Handle candidates
     }
 });
 ```
 
-Finally, pass the task instance to the `SaltyRTCBuilder` through the
-`.usingTasks(...)` method.
+### Data Channel Crypto Context
+
+The task provides another security layer for data channels which can be
+leveraged by usage of a `DataChannelCryptoContext` instance. To retrieve such
+an instance, call:
+
+```java
+final DataChannelCryptoContext context = task.getCryptoContext(dataChannel.id);
+```
```

Comment:
Should we mention somewhere that the crypto context should only be retrieved once? Should we maybe even return a warning or error if the user calls `getCryptoContext` with the same data channel id twice?

When trying to get this code to run, I initially thought that the state was inside the task and that `getCryptoContext` is just a getter that can be called multiple times. That was wrong of course, but the mistake could happen again to others.

Maybe the method could also be called `createCryptoContext` instead, to make it clear that new state is created.

Reply:
> Should we mention somewhere that the crypto context should only be retrieved once?

It's actually fine to do so since the cookie is different every time.

> When trying to get this code to run, I initially thought that the state was inside the task and that `getCryptoContext` is just a getter that can be called multiple times. That was wrong of course, but the mistake could happen again to others.
> 
> Maybe the method could also be called `createCryptoContext` instead, to make it clear that new state is created.

I'll rename it. :+1:

Reply:
> It's actually fine to do so since the cookie is different every time.

It's fine security-wise, but decrypting will fail.

But since you renamed the method (`createCryptoContext`) it should be obvious that creating another crypto context will not result in the same data being used, so I'd consider this resolved.